contracts/Tokens/VAI/VAIController.sol

```diff
@@ -0,0 +1,759 @@
+
+    /// @notice Emitted when VAI float rate is changed
+    event NewVAIFloatRate(uint256 indexed oldFloatRateMantissa, uint256 indexed newFlatRateMantissa);
+
+    /// @notice Emitted when VAI receiver address is changed
+    event NewVAIReceiver(address indexed oldReceiver, address indexed newReceiver);
+
+    /// @notice Emitted when VAI mint cap is changed
+    event NewVAIMintCap(uint256 indexed oldMintCap, uint256 indexed newMintCap);
+
+    /// @notice Emitted when VAI token address is changed by admin
+    event NewVAIToken(address indexed oldVaiToken, address indexed newVaiToken);
+
+    /// @notice Emitted when VAIMintRate is changed
+    event NewVAIMintRate(uint256 indexed oldVAIMintRate, uint256 indexed newVAIMintRate);
+
+    /// @notice Thrown when action is restricted to the caller
+    error CallerNotAuthorized();
+
+    /// @notice Thrown when treasury percent is overflowed
+    error TreasuryPercentOverflow();
+
+    /// @notice Thrown when trying to perform an action that is paused
+    error ActionPaused();
+
+    /// @notice Thrown when further minting of VAI tokens is restricted due to reaching the MintCap
+    error MintCapReached();
+
+    /// @notice Thrown when mint enables only for prime holders is 'false' and Prime address is not set
+    error ToggleConditionNotMet();
+
+    /// @notice Thrown when there is an insufficient balance of mintable VAI tokens
+    error InsufficientMintableVAIBalance();
+
+    /**
+     * @notice Initializes the VAI Controller
+     *  @param accessControlManager_ Address of the AccessControlManager contract
+     */
+    function initialize(address accessControlManager_) external initializer {
+        __Ownable2Step_init();
+        __AccessControlled_init_unchained(accessControlManager_);
+        __ReentrancyGuard_init();
+
+        VAIMintIndex = INITIAL_VAI_MINT_INDEX;
+        accrualBlockNumber = _getBlockNumber();
+        mintCap = type(uint256).max;
+    }
+
+    /**
+     * @notice The mintVAI function mints and transfers VAI from the protocol to the user, and adds a borrow balance
+     * The amount minted must be less than the user's Account Liquidity and the mint VAI limit
+     * @param mintVAIAmount The amount of the VAI to be minted
+     * @custom:event MintFee emits on success
+     * @custom:event MintVAI emits on success
+     * @custom:error ZeroAddressNotAllowed is thrown when comptroller address is zero
+     * @custom:error ZeroValueNotAllowed is thrown if mintVAIAmount is 0
+     * @custom:error ActionPaused is thrown when mint action is disabled
+     * @custom:error MintCapReached is thrown when mint limit is reached
+     * @custom:error InsufficientMintableVAIBalance is thrown when VAI mint amount is more than account mintable VAI amount
+     */
+    // solhint-disable-next-line code-complexity
+    function mintVAI(uint256 mintVAIAmount) external nonReentrant {
+        ensureNonzeroAddress(address(comptroller));
```

Comment:
While using `ensureNonzeroAddress` in the core pool `VAIController`, it would revert the transaction if the comptroller address is zero. However, in general, I observed that transactions are not reverted in core pool contracts.